python/lib/core/dmod/core/meta_data.py

```diff
@@ -184,6 +188,38 @@ class DataFormat(PydanticEnum):
     T_ROUTE_CONFIG = (13, {StandardDatasetIndex.DATA_ID: None, StandardDatasetIndex.HYDROFABRIC_ID: None}, None, False)
     """ Format for t-route application configuration. """
 
+    NGEN_GEOPACKAGE_HYDROFABRIC_V2 = (14,
+                                      {StandardDatasetIndex.CATCHMENT_ID: "divide_id",
+                                       StandardDatasetIndex.HYDROFABRIC_ID: None,
+                                       StandardDatasetIndex.HYDROFABRIC_REGION: None,
+                                       StandardDatasetIndex.HYDROFABRIC_VERSION: None},
+                                      {"fid": int, "divide_id": str, "geom": Any, "toid": str, "type": str,
+                                       "ds_id": float, "areasqkm": float, "id": str, "lengthkm": float,
+                                       "tot_drainage_areasqkm": float, "has_flowline": bool},
+                                      )
+    """ GeoPackage hydrofabric format v2 used by NextGen (id is catchment id). """
+
+    EMPTY = (15, {}, None, False)
```

Comment:
How do these interact with the `Dataset`'s `DataCategory`? At least, `EMPTY` and `GENERIC` seem related to that concept.

Reply:
Hmm ... I don't think there's any coupling, strictly speaking, between categories and either formats or domains.  For this in particular, I think that's fine:  it's reasonable to have a dataset in any category for which the data coverage is defined to be either empty or unknown. 

More generally, I can already see things with categories that need tweaking (the notion of an `OUTPUT` category is probably not valid, if we start to think of evaluation task needing modeling "output" as input, or forcing regridding jobs producing as "output" something belonging to `FORCING` category).  And we may not be using category sufficiently and/or consistently throughout our data modeling.  

Did you have something specific in mind when you noted this, or just a general concern?

Reply:
There definitely is conceptually, just not one that's explicitly defined somewhere.  

I haven't tested that scenario, but I expect it would allow the ``HYDROFABRIC`` category dataset to have a defined domain with the ``AORC_CSV`` format.   I'm pretty sure the same thing would happen if the user tried to provided the domain manually when creating a dataset, though.  So this is definitely a problem, but outside the scope of the automatic detection work.  I've opened #567 specifically for it.